js/coindcx.js

```diff
@@ -0,0 +1,596 @@
+            parsedData.push ([price, amount]);
+        }
+        return parsedData;
+    }
+
+    async fetchBalance (params = {}) {
+        // https://coindcx-official.github.io/rest-api/?javascript#get-balances
+        await this.loadMarkets ();
+        const request = {
+            'timestamp': this.milliseconds (),
+        };
+        const response = await this.privatePostExchangeV1UsersBalances (this.extend (request, params));
+        const result = { 'info': response };
+        for (let i = 0; i < response.length; i++) {
+            const balance = response[i];
+            const currencyId = this.safeString (balance, 'currency');
+            const code = this.safeCurrencyCode (currencyId);
+            if (!(code in result)) {
+                const account = this.account ();
+                const free = this.safeFloat (balance, 'balance');
+                const used = this.safeFloat (balance, 'locked_balance');
+                const total = parseFloat (free + used);
+                account['free'] = free;
+                account['used'] = used;
+                account['total'] = total;
+                result[code] = account;
+            }
+        }
+        return result;
+    }
+
+    async fetchOrder (id, symbol = undefined, params = {}) {
+        // https://coindcx-official.github.io/rest-api/?javascript#account-trade-history
+        if (id === undefined) {
+            throw new ArgumentsRequired (this.id + ' fetchOrder requires an `id` argument');
+        }
+        await this.loadMarkets ();
+        const request = {
+            'id': id,
+        };
+        const response = await this.privatePostExchangeV1OrdersStatus (this.extend (request, params));
+        return this.parseOrder (response);
+    }
+
+    async fetchOpenOrders (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        if (symbol === undefined) {
+            throw new ArgumentsRequired (this.id + ' fetchOpenOrders requires a `symbol` argument');
+        }
+        await this.loadMarkets ();
+        if (!(symbol in this.markets)) {
+            throw new ExchangeError (this.id + ' has no symbol ' + symbol);
+        }
+        const market = this.market (symbol);
+        const request = {
+            'market': this.safeValue (market, 'id'),
+            'timestamp': this.milliseconds (),
+        };
+        const response = await this.privatePostExchangeV1OrdersActiveOrders (this.extend (request, params));
+        const orders = this.safeValue (response, 'orders', []);
+        return this.parseOrders (orders, market, since, limit);
+    }
+
+    async createOrder (symbol, type, side, amount, price = undefined, params = {}) {
+        // https://coindcx-official.github.io/rest-api/?javascript#new-order
+        if (symbol === undefined) {
+            throw new ArgumentsRequired (this.id + ' createOrder requires a `symbol` argument');
+        }
+        if (type === undefined) {
```

Comment:
We don't need to check the mandatory arguments here or anywhere else throughout this file. Only optional arguments should be checked.